extensions/shared/premium-blocks/render-premium-icon.js

```diff
@@ -0,0 +1,15 @@
+/**
+ * Internal dependencies
+ */
+import PaidSymbol from './paid-symbol';
+
+const renderPremiumIcon = icon => {
+	if ( icon?.src ) {
+		icon.src.props.children = [ icon.src.props.children, <PaidSymbol /> ];
+	} else if ( icon?.props?.children ) {
+		icon.props.children = [ icon.props.children, <PaidSymbol /> ];
+	}
+	return icon;
+};
```

Comment:
Can we add tests to this function to make sure `icon.src.props.children` and `icon.props.children` are set correctly? Under which circumstance does an icon have its props within `src`?

Reply:
Thanks for pointing this out @obenland 

Some blocks (eg eventbrite) 

https://github.com/Automattic/jetpack/blob/5f22db71ba41a9301bc5015e74f1f406d302afa9/extensions/blocks/eventbrite/index.js#L39-L49

have their icon declared within a `src` so they can also use other properties as `foreground`. 
Added `jsdoc` and refactored the `src` section to use `cloneElement` for the moment.
I will add some simple tests (for that two specific cases) making sure we don't break some functionality in the near future.

Reply:
Great, thanks so much!

Reply:
@obenland , added the 2 basic tests
https://github.com/Automattic/jetpack/pull/16611/commits/b17a7cfe4564be0bb8ad53dea4695bc3f930a70b

you may run them with `yarn test-extensions`